### PR TITLE
docs: add a debug.md

### PR DESCRIPTION
**Pull request by codyoss**
This is a doc that can grow overtime. It is meant to be a resource we can share with our users when they open issues and are trying to track down bugs.

### CODE REVIEW

File: debug.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+## gRPC based clients
+
+### Try setting grpc-go's debug variables
+
+Try setting the following environment variables for grpc-go:
+`GRPC_GO_LOG_VERBOSITY_LEVEL=99` `GRPC_GO_LOG_SEVERITY_LEVEL=info`. These are
+good for diagnosing connection level failures. For more information please see
+[grpc-go's debug documentation](https://pkg.go.dev/google.golang.org/grpc/examples/features/debugging#section-readme).
+
+### Add in your own logging with a gRPC interceptors
```
[the remark this answers is not in the document]
Comment (codyoss) [reply debug.md:95]: I think that is a good call out. I will add a disclaimer to the sections the log request payloads

File: debug.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+func main() {
+    ctx := context.Background()
+
+    // Create a transport with authentication built-in detected with
+    // [ADC](https://google.aip.dev/auth/4110). Note you will have to pass any
+    // required scoped for the client you are using.
+    trans, err := htransport.NewTransport(ctx,
+        http.DefaultTransport,
```
Comment (tritone) [inline debug.md:70]: Note that this leaves out the perf tuning on the transport that we do in google-api-go-client. I've struggled to communicate this with customers in the past, any thoughts?

File: debug.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tritone) [reply debug.md:101]: +1, this should be a top level thing.

**Review by noahdietz — APPROVED** [review]

File: debug.md (hunk 2 above)
Comment (codyoss) [reply debug.md:70]: I think that is fine for the scope of this document. I would say these are one-off type of things users should add to their code when required. I would not expect this to be something that is long-lived. So I think it is fine that we gloss over those details here. Thoughts?

Comment (tritone) [reply debug.md:70]: If that's the case I would want to clarify that these are intended for local debugging and not for production use. I have seen users put this kind of logic into production to try to catch transient issues for example. It's problematic if they accidentally hurt their performance while doing that

Comment (codyoss) [reply debug.md:73]: Added a general top-level warning about perf and such

File: debug.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
+# Debugging tips and tricks
+
+While working with the Go Client libraries you may run into some situations
+where you need a deeper level of understanding about what is going on in order
+to solve your problem. Here are some tips and tricks that you can use in these
+cases. Note that many of the tips in this document will have a performance
```
Comment (tritone) [inline debug.md:6]: Maybe bold *Note* as well :)

**Review by tritone — APPROVED** [review]

File: debug.md (hunk 3 above)
Comment (codyoss) [reply debug.md:6]: ack

**Pull request merged**